packages/puppeteer-core/src/node/ScreenRecorder.ts

```diff
@@ -174,7 +199,14 @@ export class ScreenRecorder extends PassThrough {
     );
   }
 
-  #getFormatArgs(format: 'webm' | 'gif') {
+  #getFormatArgs(
+    format: 'webm' | 'gif' | 'mp4',
+    fps: number | 'fps=source_fps',
+    loop: number | typeof Infinity | boolean,
```

Comment:
Why do we need `typeof Infinity`, 
`const test: number = Infinity;` work in TypeScript.

Reply:
> Why do we need `typeof Infinity`

Suppose I just wanted to emphasize `Infinity` as a nice, semantic option. But simply `loop?: number | Infinity;` is a type error (see related [issue](https://github.com/microsoft/TypeScript/issues/32277)…)

> We should allow only `false` and not `boolean`, as having loop true does not make sense without a number

`loop = true` implies an infinite loop to me, but I guess it's subjective. I went with the below suggestion anyway…

> make this only a number, and it will not be looped if its undefined.